array_test.go

```diff
@@ -465,9 +466,7 @@ func TestArrayInsertAndGet(t *testing.T) {
 		for i := uint64(1); i < arrayCount; i += 2 {
 			v := test_utils.Uint64Value(i)
 
-			expectedValues = append(expectedValues, nil)
-			copy(expectedValues[i+1:], expectedValues[i:])
-			expectedValues[i] = v
+			expectedValues = slices.Insert[[]atree.Value, atree.Value](expectedValues, int(i), v)
```

Comment:
Here and below: The explicit type arguments can probably be inferred
```suggestion
			expectedValues = slices.Insert(expectedValues, int(i), v)
```